saleor/graphql/order/types.py

```diff
@@ -1822,6 +1825,34 @@ def resolve_checkout_id(root: models.Order, _info):
             return graphene.Node.to_global_id("Checkout", root.checkout_token)
         return None
 
+    @staticmethod
+    def __resolve_references(roots: List["Order"], info):
+        requestor = get_user_or_app_from_context(info.context)
+        requestor_has_access_to_all = has_one_of_permissions(
+            requestor, [OrderPermissions.MANAGE_ORDERS]
+        )
+
+        roots_ids = []
+        for root in roots:
+            roots_ids.append(root.id)
+
+        if requestor:
+            qs = resolve_orders(
+                info,
+                requestor_has_access_to_all=requestor_has_access_to_all,
+                requesting_user=info.context.user,
+                ids=roots_ids,
+            )
+        else:
+            qs = models.Order.objects.none()
+
+        orders = {}
+        for order in qs:
+            global_id = graphene.Node.to_global_id("Order", order.id)
+            orders[global_id] = order
+
+        return [orders.get(root_id) for root_id in roots_ids]
```

Comment:
We have `resolve_federation_references` helper function which solves this part, I think we could reuse it here.

Reply:
Other than that it looks good IMO :)